GitCommands/Settings/IViewSetting.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Data.SqlTypes;
+
+namespace GitCommands
+{
+    public interface IViewSetting
```

Comment:
I feel this is taking is further away from the [Options pattern](https://learn.microsoft.com/dotnet/core/extensions/options). I vaguely remember ivangrek did (or was working towards) a POC that converted our settings to `IOptons<T>`...

Reply:
The only "challenge" I see is that `IOption<TOptions>` expects `TOptions` being a `class`. Our options are of type `bool`, `int`, `Enum` (and `string`).

![image](https://user-images.githubusercontent.com/36601201/235245513-2605d304-0a3f-4034-b07e-e0bc375dce2f.png)

Let's continue this discussion in the base PR #10851.